File: test/csrc/com/xuggle/xuggler/ErrorTest.cpp
/*******************************************************************************
 * Copyright (c) 2008, 2010 Xuggle Inc.  All rights reserved.
 *  
 * This file is part of Xuggle-Xuggler-Main.
 *
 * Xuggle-Xuggler-Main is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Xuggle-Xuggler-Main is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with Xuggle-Xuggler-Main.  If not, see <http://www.gnu.org/licenses/>.
 *******************************************************************************/

/*
 * ErrorTest.cpp
 *
 *  Created on: Mar 20, 2009
 *      Author: aclarke
 */

#include "ErrorTest.h"
#include <com/xuggle/xuggler/FfmpegIncludes.h>
#include <com/xuggle/ferry/RefPointer.h>
#include <com/xuggle/ferry/Logger.h>

#include <com/xuggle/xuggler/IError.h>
#include <cstring>

using namespace com::xuggle::ferry;

using namespace VS_CPP_NAMESPACE;

VS_LOG_SETUP(VS_CPP_PACKAGE);

ErrorTest :: ErrorTest()
{
}

ErrorTest :: ~ErrorTest()
{
}

void
ErrorTest :: testErrorCreation()
{
  RefPointer<IError> error = IError::make(AVERROR_IO);
  
  VS_TUT_ENSURE("made error", error);
  
  const char* description = error->getDescription();
  
  VS_TUT_ENSURE("got description", description && *description);

  VS_TUT_ENSURE_EQUALS("should map correctly", 
      IError::ERROR_IO,
      error->getType());
  
  error = IError::make(IError::ERROR_IO);
  VS_TUT_ENSURE("made error", error);
    
  description = error->getDescription();

  VS_TUT_ENSURE("got description", description && *description);

  VS_TUT_ENSURE_EQUALS("should map correctly", 
      AVERROR_IO,
      error->getErrorNumber());
  
}

File: test/src/com/xuggle/xuggler/ErrorTest.java
/*
 * Copyright (c) 2008, 2009 by Xuggle Incorporated.  All rights reserved.
 * 
 * This file is part of Xuggler.
 * 
 * You can redistribute Xuggler and/or modify it under the terms of the GNU
 * Affero General Public License as published by the Free Software
 * Foundation, either version 3 of the License, or (at your option) any
 * later version.
 * 
 * Xuggler is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public
 * License for more details.
 * 
 * You should have received a copy of the GNU Affero General Public License
 * along with Xuggler.  If not, see <http://www.gnu.org/licenses/>.
 */

package com.xuggle.xuggler;
import static org.junit.Assert.*;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

public class ErrorTest
{

  @Before
  public void setUp() throws Exception
  {
  }

  @After
  public void tearDown() throws Exception
  {
  }

  @Test
  public void testGetType()
  {
    IError err = IError.make(IError.Type.ERROR_EOF);
    assertEquals(IError.Type.ERROR_EOF, err.getType());
  }

  @Test
  public void testGetDescription()
  {
    IError err = IError.make(IError.Type.ERROR_EOF);
    assertNotNull(err.getDescription());
    assertTrue(err.getDescription().length()>0);
  }

  @Test
  public void testGetErrorNumber()
  {
    IError err = IError.make(IError.Type.ERROR_EOF);
    assertTrue(err.getErrorNumber()<0);
  }

  @Test
  public void testMake1()
  {
    assertNull(IError.make(1)); // should fail with positive number
  }
  
  @Test
  public void testMake2()
  {
    assertNotNull(IError.make(IError.Type.ERROR_EOF));
  }

}
